File: ios/Utils/Mapper/RewardMapper.swift
import Foundation
import YandexMobileAds

private let typeKey = "type"
private let amountKey = "amount"

class RewardMapper {
    func map(_ reward: Reward) -> [String: Any] {
        let result: [String: Any] = [
            typeKey: reward.type,
            amountKey: reward.amount
        ]
        return result
    }
}

File: android/src/main/java/com/yandexmobileads/util/mapper/RewardMapper.kt
package com.yandexmobileads.util.mapper

import com.yandex.mobile.ads.rewarded.Reward

class RewardMapper {
    fun map(reward: Reward): Map<String, Any> = mapOf(
        KEY_TYPE to reward.type,
        KEY_AMOUNT to reward.amount
    )

    companion object {
        private const val KEY_TYPE = "type"
        private const val KEY_AMOUNT = "amount"
    }
}
